base/errorshow.jl

```diff
@@ -165,6 +165,42 @@ end
 
 typesof(args...) = Tuple{Any[ Core.Typeof(a) for a in args ]...}
 
+function print_with_compare(io::IO, a::DataType, b::DataType, color)
+    if a.name === b.name
+        Base.show_type_name(io, a.name)
+        n = min(length(a.parameters), length(b.parameters))
+        print(io, '{')
+        for i = 1:n
+            print_with_compare(io::IO, a.parameters[i], b.parameters[i], color)
+            i < n && print(io, ',')
+        end
+        print(io, '}')
+    else
+        printstyled(io, a; color=color)
+    end
+end
+
+function print_with_compare(io::IO, a, b, color)
+    if a === b
+        print(io, a)
+    else
+        printstyled(io, a; color=color)
+    end
+end
+
+function show_convert_error(io::IO, ex::MethodError, arg_types_param)
+    # See #13033
+    T = striptype(ex.args[1])
+    if T === nothing
+        print(io, "First argument to `convert` must be a Type, got ", ex.args[1])
+    else
+        printstyled(io, "Cannot `convert` an object of type ")
+        print_with_compare(io, arg_types_param[2], T, :green)
+        printstyled(io, " to an object of type ")
```

Comment:
```
printstyled(io, "Cannot `convert` an object of type \n")
print_with_compare(io, arg_types_param[2], T, :green)
printstyled(io, "\nto an object of type \n")
```
Vertical alignment should help with the eyesore.

Reply:
I've adjusted this as suggested, and I think it's good for the long errors, since it aligns printing so it's easier to scan both simultaneously, but it looks odd for short types:
<img width="394" alt="Screen Shot 2019-10-03 at 3 44 22 AM" src="https://user-images.githubusercontent.com/1291671/66108241-24fb2180-e590-11e9-918a-ba62ea04fefb.png">

perhaps we should continue to print on one line if the typenames don't match at all (and in that case probably just omit the type parameters entirely).